impeller/typographer/rectangle_packer.cc

```diff
@@ -158,11 +165,11 @@ void SkylineRectanglePacker::addSkylineLevel(int skylineIndex,
     }
   }
 
-  // merge skyline_s
-  for (int i = 0; i < ((int)skyline_.size()) - 1; ++i) {
+  // merge skylines
+  for (auto i = 0u; i < skyline_.size() - 1; ++i) {
     if (skyline_[i].y_ == skyline_[i + 1].y_) {
       skyline_[i].width_ += skyline_[i + 1].width_;
-      skyline_.erase(std::next(skyline_.begin(), i));
+      skyline_.erase(skyline_.begin() + i + 1);
```

Comment:
I always have a tough time reading the stuff that returns an iterator you must use to modify the container.

But, this is an off by one error right? That is `std::next(begin(), i) + 1` would have been the same thing?

Reply:
Yeah, looking at the diff I think this was the bug.

Reply:
Thanks. I was wondering what was different.